### ISSUE TITLE
Report Viewer broken

### ISSUE THREAD

**Issue by sebinside** [jplag/JPlag#571]
Unfortunately, the current version of the comparison view of the (deployed) version of the report viewer is broken. This has already been discussed in the last PR #535 and now has independently been found by @tsaglam. An example zip is [attached](https://github.com/jplag/JPlag/files/9275359/result_demo1-rel.zip). The error looks similar with something broken in the `store.ts` file:

```
TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
    at Function.from (<anonymous>)
    at Object.<anonymous> (store.ts:32:26)
    at Pt.getComparison (ComparisonFactory.ts:5:40)
    at setup (index.js:60:30)
    ...
```

@nestabentum Please try to look at this as soon as possible, as it affects the publicly deployed report viewer.

**Comment by nestabentum** [comment]
I will look into it as soon as possible. I took a quick look at the files you attached - the file `Test.java` in `Submissions/Timur` seems to be corrupted. It is has a size of 0kb and opening displays no contents. Is this also the case for the local version of your files or was it corrupted during the upload @tsaglam?

**Comment by tsaglam** [comment]
> Is this also the case for the local version of your files or was it corrupted during the upload @tsaglam?

Ah, sorry. Yes, there is an empty file in there, these files are old demo files and JPlag should be able to handle empty files. However, the issue also persists without the empty files (on all combinations of Windows 10 and macOS Monterey with Firefox and Chrome):

[ReportViewerProblem.zip](https://github.com/jplag/JPlag/files/9276289/ReportViewerProblem.zip)

**Comment by nestabentum** [comment]
Of course, I will add compatibility with empty files. However, I zipped up the `results` folder dropped it into the viewer and the comparison view works perfectly fine (except for the missing match indication):
<img width="695" alt="Screenshot 2022-08-07 at 09 05 13" src="https://user-images.githubusercontent.com/33717535/183279497-1ec3750e-af3f-4b39-99d9-1265616fc670.png">

I did all of this on macOS Monterey, with the built-in compression and tested the viewer in Chrome, Firefox and Safari. What program did you use for compression and on what OS did you execute the compression? @tsaglam

**Comment by tsaglam** [comment]
oops, I edited ur comment by accident!

> I did all of this on macOS Monterey, with the built-in compression and tested the viewer in Chrome, Firefox and Safari. What program did you use for compression and on what OS did you execute the compression? @tsaglam 

I used the built-in compression on macOS and 7zip on Windows. This is really odd!

<details>
<summary> HW/SW details </summary>


Browsers: 

- Firefox 103.0.1 (64-Bit)
- Chrome 103.0.5060.134 (x86_64)

</details>

**Comment by tsaglam** [comment]
I just tested it with another MacBook, which is never an ARM one. There it works, but only after rezipping the files.
This means if I zip the results on the older Intel-based MacBook and send it to the ARM one, it does not work. If I then extract the files and re-zip it, it will work.

The ARM MacBook:

<img width="215" alt="grafik" src="https://user-images.githubusercontent.com/4396919/183285733-dea3430a-8b89-4d47-8f22-a690735f53cb.png">

(with Firefox 103.0.1 64-Bit)

**Comment by dfuchss** [comment]
Maybe the way to go is to Zip the files directly in Java. Java has Zip Support maybe then the problem will not directly occur.

**Comment by nestabentum** [comment]
@tsaglam I found the issue: I falsely assumed that the submission file name would always have the same index when splitting its path, as this was always the case when I tested the viewer. The submission files in my zip always had the path `result/submissions/<submission-file>`, therefore I extracted the submission file name from splitting the path at / and accessing index 2.
The submission files in the zip you provided have the path `submission/<submission-file>`, for whatever reason. 

Consequently, I changed the file name extraction to locate the index of the `submission` folder and find the submission files by a accessing the the found index+1.


The report viewer has no problem handling empty files, nevertheless I changed how they are displayed: Instead of showing one empty line, the text "Empty File" is displayed.

<img width="723" alt="Screenshot 2022-08-07 at 12 48 27" src="https://user-images.githubusercontent.com/33717535/183287580-53d6c8ba-af44-4e59-8236-3f6c5943ec2b.png">


PR incoming.

@dfuchss this might be worth implementing. It could maybe also eliminate the need to check for the __MACOSX folder, like my current solution has to. Of course, this requires Java to not use the OS's built-in zip compression. But thinking about it, the check does not hurt as users could still unzip and zip their report with what and why ever they want to.

**Comment by sebinside** [comment]
@dfuchss @nestabentum We should totally discuss this feature in the next dev meeting as it goes in line with our updated requirements on local execution vs. deployment of the web viewer. One could even argue that (in our current world view), a zipped result should be the new default. I also already added this to the main report viewer issue #357.

**Comment by sebinside** [comment]
Closed with #573.
